include/hal/library/requester/measlib.h

```diff
@@ -0,0 +1,67 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2024 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
+ **/
+
+#ifndef REQUESTER_MEASLIB_H
+#define REQUESTER_MEASLIB_H
+
+#include "hal/base.h"
+#include "internal/libspdm_lib_config.h"
+#include "library/spdm_return_status.h"
+#include "industry_standard/spdm.h"
+
+#if LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP
+
+/**
+ * Fill HEM measurement block.
+ *
+ * @param  measurement_block          A pointer to store measurement block.
+ * @param  measurement_hash_algo      Indicates the measurement hash algorithm.
+ *                                    It must align with measurement_hash_alg
+ *                                    (SPDM_ALGORITHMS_MEASUREMENT_HASH_ALGO_*)
+ *
+ * @return measurement block size.
+ **/
+size_t libspdm_fill_measurement_hem_block (
+    spdm_measurement_block_dmtf_t *measurement_block, uint32_t measurement_hash_algo
+    );
+
+/**
+ * verify the input HEM measurement block format.
+ *
+ * @param  measurement_record         A pointer to HEM measurement block.
+ * @param  measurement_record_length  The size of HEM measurement block.
+ * @param  measurement_hash_algo      Indicates the measurement hash algorithm.
+ *                                    It must align with measurement_hash_alg
+ *                                    (SPDM_ALGORITHMS_MEASUREMENT_HASH_ALGO_*)
+ *
+ * @retval true  the input measurement_record is HEM format.
+ * @retval false the input measurement_record is not HEM format..
+ **/
+bool libspdm_verify_measurement_hem_format(
```

Comment:
why put this in HAL?

Reply:
I have moved the check to `libspdm_req_get_measurement_extension_log.c`. Thanks.